text/0008-uniform-random-tip-selection/0008-uniform-random-tip-selection.md

```diff
@@ -0,0 +1,154 @@
++ Feature name: `uniform-random-tip-selection`
++ Start date: 2020-03-09
++ RFC PR: [iotaledger/protocol-rfcs#0008](https://github.com/iotaledger/protocol-rfcs/pull/0008)
+
+# Summary
+
+Weighted Uniform Random Tip Selection on a subset enables a node to perform fast tip-selection to increase message throughput.
+The algorithm selects tips which are non-lazy in order to maximize confirmation rate.
+
+# Motivation
+
+Because of the `white-flag` confirmation algorithm, it is no longer necessary to perform complex
+tip-selection which evaluates ledger mutations while walking, therefore, a simpler and better 
+performing algorithm can be used to select tips, which in turn increases overall message throughput.
+
+In order to maximize confirmation rate however, the algorithm needs to return tips which are `non-lazy`.
+Non-lazy in this context means that a tip does not attach to a cone of messages which is too far
+in the past. Such a cone is likely to be already confirmed and does not contribute to the
+rate of newly confirmed messages when a milestone is issued.
```

Comment:
```suggestion
# Summary

Weighted Uniform Random Tip Selection on a subset enables a node to perform fast tip-selection to increase message throughput.
The algorithm selects tips which are non-lazy to maximize confirmation rate.

# Motivation

Because of the `white-flag` confirmation algorithm, it is no longer necessary to perform complex
tip-selection which evaluates ledger mutations while walking. Therefore, a more simple, better 
performing algorithm can be used to select tips, which in turn increases overall message throughput.

To maximize confirmation rate however, the algorithm needs to return tips which are `non-lazy`.
Non-lazy in this context means that a tip does not attach to a cone of messages which is too far
in the past. Such a cone is likely to be already confirmed and does not contribute to the
rate of newly confirmed messages when a milestone is issued.
```